astrocut/asdf_cutouts.py

```diff
@@ -0,0 +1,57 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""This module implements cutout functionality similar to fitscut, but for the ASDF file format."""
+
+import asdf
+import astropy
+
+def get_center_pixel(file, ra, dec):
+
+    # Get the roman 2D science image
+    f = asdf.open(file)
+
+    # Get the WCS
+    wcs = f['roman']['meta']['wcs']
+    
+    # Get the WCS header
+    header = wcs.to_fits_sip()
+    
+    # Update WCS header with some keywords that it's missing.
+    # Otherwise, it won't work with astropy.wcs tools (TODO: Figure out why. What are these keywords for?)
+    for k in ['cpdis1', 'cpdis2', 'det2im1', 'det2im2', 'sip']:
+        header[k] = 'na'
+    
+    # New WCS object with updated header
+    wcs_updated = astropy.wcs.WCS(header)
+
+    # Turn input RA, Dec into a SkyCoord object
+    coordinates = astropy.coordinates.SkyCoord(ra, dec, unit='deg')
+
+    # Map the coordinates to a pixel's location on the Roman 2d array (row, col)
+    row, col = astropy.wcs.utils.skycoord_to_pixel(coords=coordinates, wcs=wcs_updated)
+
+    return (row, col), wcs_updated
+
+
+def get_cutout(file, coords, wcs, size=20, outfile="example_roman_cutout.fits"):
+    
+    # Get the 2D science image
+    f = asdf.open(file)
+    data = f['roman']['data']
+
+    coordinates = coords
+    cutout = astropy.nddata.Cutout2D(data, position=coordinates, wcs=wcs, size=(size, size))
+
+    astropy.io.fits.writeto(outfile, data=cutout.data.value, header=cutout.wcs.to_header(), overwrite=True)
```

Comment:
```suggestion
    # Get the 2D science image
    with asdf.open(file) as f:
        data = f['roman']['data']

        coordinates = coords
        cutout = astropy.nddata.Cutout2D(data, position=coordinates, wcs=wcs, size=(size, size))

        astropy.io.fits.writeto(outfile, data=cutout.data.value, header=cutout.wcs.to_header(), overwrite=True)
```
Similar to the other comment this will help to close the file.